### ISSUE TITLE
a "&:hover" not being applied when extending config

### ISSUE THREAD

**Issue by jpanther** [tailwindlabs/tailwindcss-typography#231]
### What version of @tailwindcss/typography are you using?

v0.5.0

### What version of Node.js are you using?

v17.3.0

### What browser are you using?

Safari and Firefox

### What operating system are you using?

macOS

### Reproduction repository

https://play.tailwindcss.com/Fq1bsNWhQm?file=config

### Describe your issue

When using extend in the Tailwind config to customise the typography classes, the "&:hover" variants don't seem to be being applied in the browser. I've tried using the [most basic example in the Tailwind docs](https://tailwindcss.com/docs/typography-plugin#customizing-the-css) and even that doesn't work.

```js
module.exports = {
  theme: {
    extend: {
      typography: {
        DEFAULT: {
          css: {
            color: '#000',
            a: {
              color: '#888',
              '&:hover': {
                color: '#FF0000',
              },
            },
          },
        },
      },
    },
  },
  plugins: [
    require('@tailwindcss/typography'),
    // ...
  ],
}
```

The "Read the docs &rarr;" link appears grey as expected, but should be red on hover. 

![Screen Shot 2021-12-23 at 19 19 08](https://user-images.githubusercontent.com/4462786/147210437-52b82e25-5f7f-4e8f-9f15-7bfd6f434a59.JPG)

Inspecting the CSS in the browser shows these styles exist, but when hovering they don't take precedence over the base color definition.

![Screen Shot 2021-12-23 at 19 17 29](https://user-images.githubusercontent.com/4462786/147210159-595ddfda-0bd3-49cf-9115-2c04b0d75ee5.JPG)

**Comment by alexanderbluhm** [comment]
@jpanther  In the mean time you can also use something like `prose-a:hover:text-[#b12868]` which for me feels a bit more natural than extending the config.

**Comment by jpanther** [comment]
@alexanderbluhm Thanks, I haven't explored that method as yet... mainly because I have 60+ customisations in the config which makes it quite cumbersome to have to add them everywhere I'm using `prose`.

**Comment by btjanaka** [comment]
I've been working around this by adding `!important`, though for use cases more complicated than mine, that's sure to have unintended effects.

**Comment by rebelchris** [comment]
Following, guessing this is a bug in the latest version?

I assumed the tailwind config override was neater than applying it in each element that uses the prose class?

**Comment by sklawren** [comment]
Defining `&:hover` completely locks up the tailwind build process in Storybook and Gatsby. 

It has taken me a long time to figure out this was the cause of tailwind locking up after I upgraded to Tailwind 3.

They provide this as an example in the docs:
https://tailwindcss.com/docs/typography-plugin#customizing-the-css

This is a pretty serious bug.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

@jpanther this will be fixed by #246 and will be available in the next version.
@sklawren I tried to reproduce it with Next.js and it works just fine for me. Can you try and reproduce it and create a new issue with a reproduction attached to it? In the meantime, also try to remove your `.next` folder and restart your server.

**Comment by stevensacks** [comment]
@RobinMalfait It happens in both Storybook and Gatsby. The moment I remove the `&:hover` config, the problem goes away. If I add it back, it locks up and I have to kill -9 the process.
